### PR TITLE
Migrate layout/ to  null safety

### PR DESCRIPTION
**Pull request by amanv8060**
- Opts out every file from null-safety
- Migrates layout/ to null-safety

part of  #527 

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by amanv8060** [comment]
@guidezpl please review

File: lib/layout/text_scale.dart [hunk 1]
```diff
@@ -8,8 +8,8 @@ import 'package:flutter/material.dart';
 
 import 'package:gallery/data/gallery_options.dart';
 
-double _textScaleFactor(BuildContext context) {
-  return GalleryOptions.of(context).textScaleFactor(context);
+double? _textScaleFactor(BuildContext context) {
```
Comment (guidezpl) [inline lib/layout/text_scale.dart:11]: Let's leave this file as is for now until `GalleryOptions` is migrated

**Review by guidezpl — CHANGES_REQUESTED** [review]
```
  _    ___ _____ __  __ 
 | |  / __|_   _|  \/  |
 | |_| (_ | | | | |\/| |
 |____\___| |_| |_|  |_|
                        
```

Thanks! These PRs will be exempt from the need to add new tests.

File: lib/layout/text_scale.dart (hunk 1 above)
Comment (amanv8060) [reply lib/layout/text_scale.dart:11]: Sure , there is also analyzer issue, i will recheck on my end , I ran the tests but missed analyzer . Will make sure to run on all future pr's.

File: tool/codeviewer_cli/segment_generator.dart [hunk 2]
```diff
@@ -19,6 +22,8 @@ const _globalPrologue =
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+// @dart=2.9
```
Comment (amanv8060) [inline tool/codeviewer_cli/segment_generator.dart:25]: @guidezpl , added dart version here in generator until code_style.dart is not migrated, to fix analyzer issue

**Review by guidezpl — APPROVED** [review]

**Comment by guidezpl** [comment]
#577 will have to be merged before this PR can be

**Pull request merged**